x-pack/plugins/apm/public/components/alerting/utils/fields.tsx

```diff
@@ -98,6 +99,46 @@ export function EnvironmentField({
   );
 }
 
+export function TransactionNameField({
+  currentValue,
+  onChange,
+  serviceName,
+}: {
+  currentValue: string;
+  onChange: (value?: string) => void;
+  serviceName?: string;
+}) {
+  const label = i18n.translate('xpack.apm.alerting.fields.transaction.name', {
+    defaultMessage: 'Name',
+  });
+
+  return (
+    <PopoverExpression value={currentValue || allOptionText} title={label}>
+      <SuggestionsSelect
+        customOptions={[{ label: allOptionText, value: '' }]}
+        customOptionText={i18n.translate(
+          'xpack.apm.alerting.transaction.name.custom.text',
+          {
+            defaultMessage: 'Add \\{searchValue\\} as a new transaction name',
+          }
+        )}
+        defaultValue={currentValue}
+        fieldName={TRANSACTION_NAME}
+        onChange={onChange}
+        placeholder={i18n.translate(
+          'xpack.apm.transactionNamesSelectPlaceholder',
+          {
+            defaultMessage: 'Select transaction name',
+          }
+        )}
+        start={moment().subtract(24, 'h').toISOString()}
```

Comment:
Using url timeframe will come in a following PR/ticket?

Reply:
yes https://github.com/elastic/kibana/issues/154704